web100/pure.go

```diff
@@ -0,0 +1,469 @@
+package web100
+
+import (
+	"C"
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"strings"
+)
+
+// TODO - resolve use of Record, Snapshot, slog, snap, etc.
+
+// Need to:
+// 1. Read the header
+//   blah blah blah e.g.  "2.5.27 201001301335 net100\n"
+//   "\n"
+//   /spec\n
+//   name%20offset%20WEB100_TYPE%20length\n (separated by spaces, term by \n)
+//   ...
+//   ...
+//   \n\n
+//   /read\n
+//   \n----End-Of-Header---- -1 -1\n
+//   log_time
+//   group_name (currently "read")
+//   connection spec (binary)
+//
+//   ...
+//   \n----Begin-Snap-Data----\n
+//   ...
+// It appears that the file is expected to end at the end of a snap data
+// with no tag to indicate the end - just EOF.
+//
+//00000000  32 2e 35 2e 32 37 20 32  30 31 30 30 31 33 30 31  |2.5.27 201001301|
+//00000010  33 33 35 20 6e 65 74 31  30 30 0a 0a 2f 73 70 65  |335 net100../spe|
+
+//00000c50  4c 69 6d 43 77 6e 64 20  34 30 20 34 20 34 0a 00  |LimCwnd 40 4 4..|
+//00000c60  2d 2d 2d 2d 45 6e 64 2d  4f 66 2d 48 65 61 64 65  |----End-Of-Heade|
+//00000c70  72 2d 2d 2d 2d 20 2d 31  20 2d 31 0a 8a 57 12 59  |r---- -1 -1..W.Y|
+//00000c80  72 65 61 64 00 00 00 00  00 00 00 00 00 00 00 00  |read............|
+//00000c90  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
+//00000ca0  71 46 00 00 97 37 eb cc  bf d4 00 00 c2 74 55 cb  |qF...7.......tU.|
+//00000cb0  2d 2d 2d 2d 42 65 67 69  6e 2d 53 6e 61 70 2d 44  |----Begin-Snap-D|
+//00000cc0  61 74 61 2d 2d 2d 2d 0a  00 00 00 00 71 46 71 46  |ata----.....qFqF|
+
+const (
+	BEGIN_SNAP_DATA   = "----Begin-Snap-Data----\n"         // Plus a newline?
+	END_OF_HEADER     = "\x00----End-Of-Header---- -1 -1\n" // No newline.
+	GROUPNAME_LEN_MAX = 32
+	VARNAME_LEN_MAX   = 32
+)
+
+type VarType int
+
+const (
+	WEB100_TYPE_INTEGER VarType = iota
+	WEB100_TYPE_INTEGER32
+	WEB100_TYPE_INET_ADDRESS_IPV4
+	WEB100_TYPE_COUNTER32
+	WEB100_TYPE_GAUGE32
+	WEB100_TYPE_UNSIGNED32
+	WEB100_TYPE_TIME_TICKS
+	WEB100_TYPE_COUNTER64
+	WEB100_TYPE_INET_PORT_NUMBER
+	WEB100_TYPE_INET_ADDRESS
+	WEB100_TYPE_INET_ADDRESS_IPV6
+	WEB100_TYPE_STR32
+	WEB100_TYPE_OCTET
+	WEB100_NUM_TYPES
+)
+
+type AddrType int
+
+const (
+	WEB100_ADDRTYPE_UNKNOWN AddrType = iota
+	WEB100_ADDRTYPE_IPV4
+	WEB100_ADDRTYPE_IPV6
+	WEB100_ADDRTYPE_DNS = 16
+)
+
+const (
+	WEB100_TYPE_IP_ADDRESS = WEB100_TYPE_INET_ADDRESS_IPV4 /* Deprecated */
+	WEB100_TYPE_UNSIGNED16 = WEB100_TYPE_INET_PORT_NUMBER  /* Deprecated */
+)
+
+var Web100Sizes = [WEB100_NUM_TYPES + 1]byte{
```

Comment:
A mapping type would be clearer, but I freely admit that the clarity would come with a computational hit in one of our tightest loops and so leave it up to your judgement.  Perhaps a function that is just a big switch?